modules/data-entry/src/main/frontend/src/vocabQuery/InfoBrowser.jsx

```diff
@@ -0,0 +1,205 @@
+import QueryStyle from "./queryStyle.jsx";
+
+// Component that renders a vocabulary info box and browser.
+//
+function InfoBrowser(props) {
+  const { browserOpen, onClose, infoPath, infoButtonRefs, infoboxRef, browserRef, classes } = props;
+
+  const [termInfoVisible, setTermInfoVisible] = useState(false);
+  const [term, setTerm] = useState({});
+
+  const [browseID, setBrowseID] = useState("");
+  const [browsePath, setBrowsePath] = useState("");
+  const [browserOpened, setBrowserOpened] = useState(!!browserOpen);
+
+  const [vocab, setVocab] = useState({});
+
+  const [snackbarVisible, setSnackbarVisible] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState('');
+
+  // Strings used by the info box
+  const [infoAboveBackground, setInfoAboveBackground] = useState(false);
+  const [infoAnchor, setInfoAnchor] = useState(null);
+
+  const [buttonRefs, setButtonRefs] = useState(infoButtonRefs || {});
+  const [noResults, setNoResults] = useState(false);
+
+  useEffect(() => {
+    infoPath ? getInfo(infoPath) : setTermInfoVisible(false);
+  }, [infoPath])
+
+
+  // Event handler for clicking away from the info box or browser while they are open
+  let clickAwayInfo = (event) => {
+    if ( browserRef?.current?.contains(event.target)
+         || infoboxRef?.current?.contains(event.target)) {
+      return;
+    }
+
+    setTermInfoVisible(false);
+    setInfoAboveBackground(false);
+  }
+
+  // Register a button reference that the info box can use to align itself to
+  let registerInfoButton = (id, node) => {
+    // List items getting deleted will overwrite new browser button refs, so
+    // we must ignore deregistration events
+    if (node) {
+      buttonRefs[id] = node;
+    }
+  }
+
+  // Grab information about the given ID and populate the info box
+  let getInfo = (path) => {
+    // If we don't yet know anything about our vocabulary, fill it in
+    var vocabPath = `${path.split("/").slice(0, -1).join("/")}.json`;
+    if (vocab.path != vocabPath) {
+      var url = new URL(vocabPath, window.location.origin);
+      MakeRequest(url, parseVocabInfo);
+    }
+
+    var url = new URL(path + ".info.json", window.location.origin);
+    MakeRequest(url, showInfo);
+  }
+
+  let parseVocabInfo = (status, data) => {
+    if (status === null) {
+      setVocab( { acronym: data["identifier"] || "",
+                  url: data["website"] || "",
+                  description: data["description"] || "",
+                  path: data["@path"]
+                } );
+    } else {
+      logError("Failed to search vocabulary details");
+    }
+  }
+
+  // callback for getInfo to populate info box
+  let showInfo = (status, data) => {
+    if (status === null) {
+      var typeOf = [];
+      if ("parents" in data) {
+        typeOf = data["parents"].map(element =>
+          element["label"] || element["name"] || element["identifier"] || element["id"]
+        ).filter(i => i);
```

Comment:
What is the purpose of this `filter`? It doesn't seem necessary since `map` already makes copies.

Can this whole block actually be replaced with:

```suggestion
      data["parents"]?.map(p => p["label"] || p["name"] || p["identifier"] || p["id"]) || []
```

and moved down to line 119, for consistency with the other term entries?